### PR TITLE
Checkbox 705/fix stress ng parts

### PR DESCRIPTION
**Pull request by diohe0311**
## Description
All the snapcraft.yaml in checkbox-core-snap needs to update due to the build environment and dependencies of stress-ng is updated. More precisely, stress-ng adds flags to specifies stack protection method, and check the security of formatted strings. So checkbox-core-snap needs to be in sync with it.

According to [this forum](https://forum.snapcraft.io/t/build-snaps-with-hardened-toolchain-by-default/5444), when using `Wl` to make gcc inform what parameters should linker be dealing with, it should be writing as `-Wl`. Which is different from [upstream](https://github.com/ColinIanKing/stress-ng/blob/master/snap/snapcraft.yaml#L19).

## Resolved issues
https://warthogs.atlassian.net/browse/CHECKBOX-705

## Tests
Precondition: set up a LXD container
1. `lxc launch ubuntu:20.04 ubuntu20` # "ubuntu20" could be what ever name you want
2. `lxc exec ubuntu20 -- apt update`

Container for ubuntu16.04 has some extra steps because of [CGroup version issue](https://discuss.linuxcontainers.org/t/error-the-image-used-by-this-instance-requires-a-cgroupv1-host-system-when-using-clustering/13885):
1. `lxc launch ubuntu:16.04 ubuntu16`
2. `lxc exec ubuntu16 -- apt update`
3. `vi /etc/default/grub`
4. find `GRUB_CMDLINE_LINUX` and add `“systemd.unified_cgroup_hierarchy=false"`
5. `update-grub`
6. reboot

Build a snap and install into the container
1. `cd checkbox/checkbox-core-snap/`
2. `./prepare.sh series20`
3. `cd series20`
4. `snapcraft --use-lxd`
5. `lxc file push checkbox20_2.8_amd64.snap ubuntu20/root/`
6. `lxc exec ubuntu20 -- snap install checkbox20_2.8_amd64.snap --dangerous`
7. `lxc exec ubuntu20 -- bash` # enter the container
8. `sudo snap install checkbox --classic --channel=20.04/stable`
9. `checkbox.shell` # enter the checkbox shell
10. `type stress-ng`
11. `stress-ng -V`
12. check if output matchs:
```
root@ubuntu20:~# type stress-ng
stress-ng is /snap/checkbox20/current/usr/bin/stress-ng
root@ubuntu20:~# stress-ng -V
stress-ng, version 0.15.09 (gcc 9.4.0, x86_64 Linux 5.19.0-42-generic) 💻🔥
```

If the current snapcraft.yaml is different from the previous one, you need to run `snapcraft clean` first. Otherwise, the stress-ng part would be skipped during the build because it has already been built with the previous settings.

### CODE REVIEW

**Comment by pieqq** [comment]
Hi! I just tried to build the snap from my machine for series18, and I get the following error:

```
LD stress-ng
cc: error: Wl,-z,relro: No such file or directory
Makefile:561: recipe for target 'stress-ng' failed
make: *** [stress-ng] Error 1
Failed to run 'make -j2 STATIC=1' for 'stress-ng': Exited with code 2.
Verify that the part is using the correct parameters and try again.
```

Does it work on your side?

**Review by yphus — CHANGES_REQUESTED** [review]
[libipsec-mb-dev](https://launchpad.net/ubuntu/+source/intel-ipsec-mb) and libxxhash-dev are not available for 16.04 (xenial). I suggest to revert the tag to **v0.15.07** in `checkbox-core-snap/series16/snap/snapcraft.yaml` and not add new build dependencies to series16.
Let's even add a comment next to the tag to avoid future updates so that **v0.15.07** is kept frozen.

**Review by pieqq — CHANGES_REQUESTED** [review]
Good job figuring out how to build stress-ng with the proper LDFLAGS!

I built a checkbox18 snap from this branch. During prime stage, I got the following errors:

```
Priming stress-ng 
+ snapcraftctl prime
The 'stress-ng' part is missing libraries that are not included in the snap or base. They can be satisfied by adding the following entry for this part
stage-packages:
- libjpeg-turbo8
- libjudydebian1
- libsctp1
- libxxhash0
Priming acpi-tools 
+ snapcraftctl prime
Priming checkbox-support 
+ snapcraftctl prime
The 'checkbox-support' part is missing libraries that are not included in the snap or base. They can be satisfied by adding the following entries to the existing stage-packages for this part:
- libbluetooth3
Priming checkbox-ng 
+ snapcraftctl prime
Priming checkbox-provider-resource 
+ snapcraftctl prime
Priming checkbox-provider-base 
+ snapcraftctl prime
This part is missing libraries that cannot be satisfied with any available stage-packages known to snapcraft:
- libGLU.so.1
- libglut.so.3
- libgpm.so.2
- libslang.so.2
These dependencies can be satisfied via additional parts or content sharing. Consider validating configured filesets if this dependency was built.
```

but the snap file was built regardless.

However, when I tried to install it in a 18.04 container...

```
root@bionic:~# snap install checkbox18_2.8_amd64.snap --dangerous
2023-06-30T08:25:18Z INFO Waiting for automatic snapd restart...
checkbox18 2.8 installed
root@bionic:~# sudo snap install checkbox --classic --channel=18.04/stable
checkbox (18.04/stable) 2.7 from Canonical Certification Team (ce-certification-qa) installed
root@bionic:~# checkbox.shell
checkbox runtime shell, type 'exit' to quit the session
root@bionic:~# stress-ng --version
stress-ng: error while loading shared libraries: libJudy.so.1: cannot open shared object file: No such file or directory
```

we might be missing some `stage-packages` in the snapcraft.yaml definition!

I haven't checked for the other series, but it's worth checking I think.

**Review by yphus — APPROVED** [review]
LGTM

**Review by yphus — CHANGES_REQUESTED** [review]
^

**Comment by diohe0311** [comment]
@pieqq @yphus Thank you!
This error also happens in checkbox-support and checkbox-provider-base of series18:
checkbox-support:
- libbluetooth3
checkbox-provider-base:
- libGLU
- libglut
- libgpm
- libslang

it's fixed too! And the stage-packages in stress-ng of series20 & series22 has no problems, but the checkbox-provider-base of series20 is lack of

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 1]
```diff
@@ -113,6 +116,8 @@ parts:
       - libcap-dev
       - libsctp-dev
       - libjudy-dev
+      - libjudydebian1
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:119]: I don't think you need to add libjudydebian1 and libsctp1 to build-packages. If snapcraft issued a warning, the proprosal was to add the missing lib in the stage-packages instead

**Review by yphus — CHANGES_REQUESTED** [review]
See my proposal below to fix s16 where I think packages should be added to the stage and not build packages section

File: checkbox-core-snap/series16/snap/snapcraft.yaml (hunk 1 above)
Comment (diohe0311) [reply checkbox-core-snap/series16/snap/snapcraft.yaml]: Fixed it, and tested it by build snap again.

**Review by yphus — APPROVED** [review]
Tested a checkbox20 build, stress-ng works w/o any problem. The dependency problem is solved. +1

**Review by pieqq — APPROVED** [review]
Looks good to me! Thanks a lot for your efforts on this, Dio!

**Pull request merged**